File: GitToolBox/src/main/java/zielu/gittoolbox/ui/blame/BlameStatusLineData.java
package zielu.gittoolbox.ui.blame;

import com.intellij.openapi.util.Key;
import java.util.Objects;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import zielu.gittoolbox.revision.RevisionInfo;

class BlameStatusLineData {
  static final Key<BlameStatusLineData> KEY = new Key<>("GitToolBox-blame-status-line-info");

  private final String lineInfo;
  private final RevisionInfo revisionInfo;

  BlameStatusLineData(@Nullable String lineInfo, @NotNull RevisionInfo revisionInfo) {
    this.lineInfo = lineInfo;
    this.revisionInfo = revisionInfo;
  }

  @Nullable
  String getLineInfo() {
    return lineInfo;
  }

  boolean isSameRevision(RevisionInfo revisionInfo) {
    return Objects.equals(this.revisionInfo, revisionInfo);
  }
}

File: GitToolBox/src/main/kotlin/zielu/gittoolbox/ui/blame/BlameStatusLineData.kt
package zielu.gittoolbox.ui.blame

import com.intellij.openapi.util.Key
import com.intellij.openapi.util.UserDataHolder
import zielu.gittoolbox.revision.RevisionInfo

internal data class BlameStatusLineData(
  val revisionInfo: RevisionInfo,
  val lineInfo: String?
) {

  fun isSameRevision(revisionInfo: RevisionInfo): Boolean {
    return this.revisionInfo == revisionInfo
  }

  companion object {
    private val KEY = Key<BlameStatusLineData>("GitToolBox-blame-status-line-info")

    @JvmStatic
    fun get(userDataHolder: UserDataHolder): BlameStatusLineData? {
      return userDataHolder.getUserData(KEY)
    }

    @JvmStatic
    fun set(userDataHolder: UserDataHolder, editorData: BlameStatusLineData) {
      userDataHolder.putUserData(KEY, editorData)
    }

    @JvmStatic
    fun clear(userDataHolder: UserDataHolder) {
      userDataHolder.putUserData(KEY, null)
    }
  }
}
